src/loader.js

```diff
@@ -146,15 +147,99 @@
   };
 
   // 4. Registry Objects
+  // For now, registry objects are a work in progress that don't fully integrate into the rest of the code base
 
   // 4.1.1
+  function createRegistry() {
+    var registry = new RegistryPrototype();
+    registry._registry = {};
+    var iterator;
+    if (__global.Map && __global.Map.prototype && __global.Map.prototype.entries && __global.Map.prototype.keys
+         && __global.Map.prototype.values && __global.Map.prototype.get && __global.Map.prototype.set
+         && __global.Map.prototype.has && __global.Map.prototype.delete)
+      registry._registry.registryMap = new __global.Map();
+    else
+      registry._registry.registryMap = new InternalMapPolyfill();
+    // 4.4.2
+    if (__global.Symbol && __global.Symbol.iterator)
+      registry[__global.Symbol.iterator] = mapPolyfillEntriesIterator.bind(registry._registry.registryMap);
+    return registry;
+  }
+
+  // 4.2
+  function Registry() {
+    throw new Error('A registry may only be created when creating a loader');
+  }
```

Comment:
I think the wording there that it isn't meant to be instantiated indicates that it is an internal class, but it is yet to be determined how public the API is I believe. If it were to become public, the ability to run the constructor of the registry would be supported. Either way I think having the Registry as a standard class here would be close enough to the spec. It's worth keeping the implementation simple where possible I think, complexity can always be added later.